index.bs

```diff
@@ -701,17 +784,28 @@ urlPrefix: https://w3c.github.io/FileAPI/#dfn-; type: dfn;
 
 				1. If |r| is not "granted", then reject |p| with a "NotAllowedError" DOMException
 
-				1. Let |cleanItemList| be an empty sequence&lt;{{Blob}}>.
+				1. Let |itemList| and |cleanItemList| be an empty sequence&lt;{{Blob}}>.
+
+				1. For each |itemData| in |data|:
+
+					1. For each |item| in |itemData|:
+						1. If [=Promise=] to {{Blob}} in |item| is rejected, then throw "Promises to Blobs were rejected." DOMException.
+
+						1. Add the {{Blob}} in |item| to |itemList| after the promise has been resolved.
+
+					1. For each {{Blob}} |itemBlob| in |itemList|:
+
+						1. If {{Blob/type}} is not in the [=mandatory data types=] list, then reject |p| with a "Type {{Blob/type}} not supported on write." DOMException.
 
-				1. For each {{Blob}} |item| in |data|:
+						1. Let |cleanItem| be a sanitized copy of |itemBlob|.
 
-					1. Let |cleanItem| be a sanitized copy of |item|.
+							Issue: Add definition of sanitized copy.
 
-					1. If unable to create a sanitized copy, then reject |p|.
+						1. If unable to create a sanitized copy, then reject |p|.
 
-					1. Add |cleanItem| to |cleanItemList|.
+						1. Add |cleanItem| to |cleanItemList|.
 
-				1. Replace the [=system clipboard data=] with |cleanItemList|.
+					1. Replace the [=system clipboard data=] with |cleanItemList|.
```

Comment:
Something is wrong here. All cleaned items corresponding to exactly one `ClipboardItem` need to grouped together. And all groups together will replace the system clipboard data. With a note that currently only Apple-OSs support multiple clipboard items.

Reply:
Ack.